src/main.cpp

```diff
@@ -9,62 +9,85 @@
 #include "hitablelist.h"
 #include "camera.h"
 
+std::random_device d;
+std::mt19937 m{d()};
+std::uniform_real_distribution<float> dist(0.0f, 1.0f);
+
+Vec3 random_in_unit_sphere();
 
 Color ray_color(const Ray &r, Hitable *world);
 
 
 int main()
 {
-    Image i("test_antialiasing.ppm");
+    Image image("test_diffuse.ppm");
 
-    auto samples = 100;
+    auto samples = 8;
 
-    Hitable *list[2];
+    Hitable *list[3];
 
-    list[0] = new Sphere(Vec3(0.0f, 0.0f, -1.0f), 0.5f);
-    list[1] = new Sphere(Vec3(0.0f, -1000.0f, 0.0f), 1000.0f);
+    int i = 0;
+    list[i++] = new Sphere(Vec3(0.0f, -1000.0f, 0.0f), 1000.0f - 0.1f);
+    list[i++] = new Sphere(Vec3(0.0f, 0.0f, -1.0f), 0.5f);
+    list[i++] = new Sphere(Vec3(1.0f, 0.0f, -1.0f), 0.5f);
 
-    Hitable *world = new HitableList(list, 2);
+    Hitable *world = new HitableList(list, i);
 
     Camera cam;
 
     // RANDOM GENERATORS
     std::random_device d;
     std::mt19937 m{d()};
-    std::uniform_real_distribution<float> dist(0.0f, 1.0f);
+    auto max_rand_jitter = 1.0f - 1.0f / samples;
```

Comment:
The sample "radius" is processed as:
`1 - (1 / samples)`
This way, with only 1 sample, the random function will return 0, always, for each pixel (this way no jitter is generated).
With more than 1 sample, with higher values the random radius will approach 1.0, generating more smooth images, without having huge gaps between samples.